src/pudl/transform/classes.py

```diff
@@ -85,44 +85,53 @@ class RenameColumns(TransformParams):
     columns: dict[str, str] = {}
 
 
-class RemoveInvalidRows(TransformParams):
-    """Defines how to identify invalid rows to drop."""
+class InvalidRows(TransformParams):
+    """Pameters that identify invalid rows to drop."""
 
     invalid_values: list
-    cols_to_check: list[str] | None = None
-    cols_to_not_check: list[str] | None = None
+    """A list of values that should be considered invalid in the selected columns."""
+
+    required_valid_cols: list[str] | None = None
+    """The list of columns passed into :meth:`pd.filter` as the ``items`` argument."""
+
+    allowed_invalid_cols: list[str] | None = None
+    """The list of *not* to search for valid values to preserve.
+
+    Used to construct an ``items`` argument for :meth:`pd.filter`. This is useful if a
+    table is wide and specifcying all ``required_valid_cols`` would be tedious.
+    """
+
     like: str | None = None
+    """A string to use as the ``like`` argument to :meth:`pd.filter`"""
+
     regex: str | None = None
+    """A regular expression to use as the ``regex`` argument to :meth:`pd.filter`."""
 
     @root_validator
-    def fliter_options(cls, values):
-        """Check if the args for ``filter`` are multually exculsive.
-
-        You input either ``cols_to_check`` or ``cols_to_not_check``, which will feed
-        into ``pandas.filter``'s ``items`` parameter.
-        """
-        num_of_non_non_values = sum(
-            x is not None
-            for x in [
-                values["cols_to_check"],
-                values["cols_to_not_check"],
+    def one_filter_argument(cls, values):
+        """Validate that only one argument is specified for :meth:`pd.filter`."""
+        num_args = sum(
+            int(bool(val))
```

Comment:
maybe you think this is more clear this way, but this was previously doing the same thing without the `int(bool())` here.

Reply:
I found the double-negatives both in the variable name and the logic used to figure out how many arguments had been provided confusing.

This would work with just the `bool(val)` but for readability I wanted to make it explicit that the (potentially non-boolean) values were getting converted to bools, and than we were going to treat the bools as numbers for counting.

This arrangement also works correctly with inputs that show up as boolean `False`, even if they aren't `None`. E.g. if someone gives an empty list for one of the lists of columns, and a non-empty list for the other, the empty list counts as a False (so a zero) and that set of inputs is actually fine for what the function is doing.